src/vs/platform/profiling/common/profilingTelemetrySpec.ts

```diff
@@ -3,26 +3,83 @@
 	totalTime: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; isMeasurement: true; comment: 'Total time of the sample' };
 	percentage: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; isMeasurement: true; comment: 'Relative time (percentage) of the sample' };
 	perfBaseline: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; isMeasurement: true; comment: 'Performance baseline for the machine' };
 	functionName: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; comment: 'The name of the sample' };
 	callstack: { classification: 'CallstackOrException'; purpose: 'PerformanceAndHealth'; comment: 'The stacktrace leading into the sample' };
+	callstackHash: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; comment: 'Hash of the stacktrace' };
 	extensionId: { classification: 'SystemMetaData'; purpose: 'PerformanceAndHealth'; comment: 'The extension for the sample (iff applicable)' };
 };
+
+type ProfilingSampleError = {
+	message: string;
+	stack: string;
+};
+
+type ProfilingSampleErrorClassification = {
+	owner: 'jrieken';
+	comment: 'Detail performance trace from a heavy sample to its callers';
+	message: { 'classification': 'CallstackOrException'; 'purpose': 'FeatureInsight'; 'comment': 'The message containing the selftime and hash of the real stacktrace' };
+	stack: { 'classification': 'CallstackOrException'; 'purpose': 'FeatureInsight'; 'comment': 'A synthetic stack which is the heaviest call to a bottom-up sample annotated with relative times' };
+};
+
+
+export interface SampleData {
+	perfBaseline: number;
+	sample: BottomUpSample;
+	extensionId: string;
+}
+
+export function reportSample(name: string, data: SampleData, telemetryService: ITelemetryService, logService: ILogService): void {
+
+	const { sample, perfBaseline, extensionId } = data;
+
+	// identify the sample trace by its locations
+	const sha1 = new StringSHA1();
+	sample.caller.forEach(c => sha1.update(c.location));
+	const callstackHash = sha1.digest();
+
+	// log a fake error with a clearer stack
+	const fakeError = new Error(`${callstackHash}|${sample.selfTime}ms`);
+	fakeError.name = 'PerfSampleError';
+	fakeError.stack = `${fakeError.message} by ${data.extensionId} in ${sample.location}\n` + sample.caller.map(c => `\t at ${c.location} (${c.percentage}%)`).join('\n');
+	logService.error(fakeError, `[perf] HEAVY function sample`);
+
+	// send telemetry event AND error
+	telemetryService.publicLog2<TelemetrySampleData, TelemetrySampleDataClassification>(`${name}.sample`, {
+		perfBaseline,
+		selfTime: sample.selfTime,
+		totalTime: sample.totalTime,
+		percentage: sample.percentage,
+		functionName: sample.location,
+		callstack: sample.caller.map(c => c.location).join('<'),
+		callstackHash,
+		extensionId
+	});
+
+	telemetryService.publicLogError2<ProfilingSampleError, ProfilingSampleErrorClassification>(`${name}.sampleError`, {
```

Comment:
@lramos15 please check if this is correct usage of error telemetry, esp if this goes to the errors page.

Reply:
This won't go to the errors site, but can still be queried in Kusto is that what you want?

Reply:
Will it have the same size constraints and suffer from data cropping like normal telemetry? I don't want that ;-)

Reply:
That is a pipeline restriction and applies to the error site as well. There's no way around that